src/stream_base-inl.h

```diff
@@ -77,7 +77,13 @@ void StreamBase::AddMethods(Environment* env,
 template <class Base>
 void StreamBase::GetFD(Local<String> key,
                        const PropertyCallbackInfo<Value>& args) {
-  StreamBase* wrap = Unwrap<Base>(args.Holder());
+  StreamBase* wrap;
+
+  // Mimic implementation of StreamBase::GetFD() and UDPWrap::GetFD().
+  ASSIGN_OR_RETURN_UNWRAP(Base,
+                          wrap,
+                          args.Holder(),
+                          args.GetReturnValue().Set(-1));
```

Comment:
The IsAlive() check a few lines down returns UV_EINVAL.  Returning -1 here is equivalent to returning UV_EPERM.